### ISSUE TITLE
Flexible Pricing: Add logic to auto-approve flexible pricing

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/mitxonline#598]
Blocked by https://github.com/mitodl/mitxonline/issues/596
Blocked by https://github.com/mitodl/mitxonline/issues/597

Code to perform the application of flexible pricing automatically. This should be similar to this code in MicroMasters: https://github.com/mitodl/micromasters/blob/master/financialaid/serializers.py#L79-L96

Feel free to copy/paste that code as a starting point, but be sure to clean it up

- Use the same conditions as we did in MicroMasters in terms of income vs. tier income threshold
- Lookup the possible tiers (e.g. what `determine_tier_program` in MicroMasters is doing) by looking them up using the `GenericForeignKey` added in https://github.com/mitodl/mitxonline/issues/597
- Depending on which `FlexiblePriceTier` is determined to be applied to the Learner:
  - Set `FlexiblePrice.flexible_price_tier` to this tier.
  - Create a `UserDiscount` record for that `User`, using the `Discount` record the `FlexiblePriceTier` points to.
- Make `DEFAULT_INCOME_THRESHOLD` an application setting that can be set via an environment variable and defaults to `75000`.

**Comment by umarmughal824** [comment]
closed as already covered in https://github.com/mitodl/mitxonline/pull/580
